FreeRTOS/iot-reference-arm-corstone3xx
# Separate ML model from NS image

Description
-----------

The idea is to move the ML model to the DDR memory region, generate a separate image for the DDR memory region. To do so the following changes are needed:

* Modify the linker scripts for all platforms to remove the RAM load address of the ML model image as it should be
loaded to DDR memory region.
* Some linker sections were renamed to be more descriptive.
* A new CMake function is introduced to extract sections from input AXF file, this function is used to extract DDR section
which contains the ML model out of the NS image to be loaded separately.

Test Steps
-----------

Validated changes in the CI, no regression observed.

Checklist:
----------



Related Issue
-----------


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
